modules/cli/src/main/java/org/apache/ignite/cli/config/ini/IniConfigManager.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.cli.config.ini;
+
+import static org.apache.ignite.cli.config.ConfigConstants.CLUSTER_URL;
+import static org.apache.ignite.cli.config.ConfigConstants.CURRENT_PROFILE;
+import static org.apache.ignite.cli.config.ConfigConstants.JDBC_URL;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import org.apache.ignite.cli.config.ConfigConstants;
+import org.apache.ignite.cli.config.ConfigInitializationException;
+import org.apache.ignite.cli.config.ConfigManager;
+import org.apache.ignite.cli.config.Profile;
+import org.apache.ignite.cli.config.ProfileNotFoundException;
+
+/**
+ * Implementation of {@link ConfigManager} based on {@link IniFile}.
+ */
+public class IniConfigManager implements ConfigManager {
+    private static final String INTERNAL_SECTION_NAME = "ignitecli_internal";
+
+    private final IniFile configFile;
+
+    private String currentProfileName;
+
+    /**
+     * Constructor.
+     *
+     * @param file ini file.
+     */
+    public IniConfigManager(File file) {
+        IniFile configFile;
+        try {
+            configFile = new IniFile(file);
+            findCurrentProfile(configFile);
+        } catch (IOException | NoSuchElementException e) {
+            configFile = createDefaultConfig();
```

Comment:
Shall we log that something went wrong during the initialization and we apply some fallback logic?